src/Workspaces/Core/Portable/Workspace/FileTextLoader.cs

```diff
@@ -58,18 +60,39 @@ public FileTextLoader(string path, Encoding? defaultEncoding)
 
         internal sealed override string FilePath => Path;
 
-        protected virtual SourceText CreateText(Stream stream, Workspace workspace)
-        {
-            var factory = workspace.Services.GetRequiredService<ITextFactoryService>();
-            return factory.CreateText(stream, DefaultEncoding);
-        }
+        internal override bool CanReloadText => !IsObsoleteCreateTextOverridden;
+
+#pragma warning disable CS0618 // Type or member is obsolete
+        private bool IsObsoleteCreateTextOverridden
+            => s_isObsoleteCreateTextOverriden.GetValue(
+                GetType(), _ => new StrongBox<bool>(new Func<Stream, Workspace, SourceText>(CreateText).Method.DeclaringType != typeof(FileTextLoader))).Value;
```

Comment:
>new Func<Stream, Workspace, SourceText>(CreateText) [](http://example.com/codeflow?start=52&length=51)

Consider improving readability of the code by surrounding delegate creation with parenthesis.